internal/cmdset/cmdset_test.go

```diff
@@ -0,0 +1,194 @@
+package cmdset
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/kyaml/fieldmeta"
+	"sigs.k8s.io/kustomize/kyaml/openapi"
+)
+
+func TestSetCommand(t *testing.T) {
+	var tests = []struct {
+		name              string
+		inputOpenAPI      string
+		input             string
+		args              []string
+		out               string
+		expectedOpenAPI   string
+		expectedResources string
+		errMsg            string
+	}{
+		{
+			name: "set replicas",
+			args: []string{"--value", "replicas=4", "--description", "hi there", "--set-by", "pw"},
+			out:  "set 1 field(s)\n",
+			inputOpenAPI: `
+apiVersion: v1alpha1
+kind: Example
+openAPI:
+  definitions:
+    io.k8s.cli.setters.replicas:
+      description: hello world
+      x-k8s-cli:
+        setter:
+          name: replicas
+          value: "3"
+          setBy: me
+ `,
+			input: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 3 # {"$kpt-set":"${replicas}"}
+ `,
+			expectedOpenAPI: `
+apiVersion: v1alpha1
+kind: Example
+openAPI:
+  definitions:
+    io.k8s.cli.setters.replicas:
+      description: hi there
+      x-k8s-cli:
+        setter:
+          name: replicas
+          value: "4"
+          setBy: pw
+          isSet: true
+ `,
+			expectedResources: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+spec:
+  replicas: 4 # {"$kpt-set":"${replicas}"}
+ `,
+		},
+		{
+			name: "openAPI list values set by flag success",
+			args: []string{"--value", "list=[10, 11]"},
+			out:  "set 1 field(s)\n",
+			inputOpenAPI: `
+kind: Kptfile
+openAPI:
+  definitions:
+    io.k8s.cli.setters.list:
+      type: array
+      x-k8s-cli:
+        setter:
+          name: list
+          listValues:
+          - 0
+ `,
+			input: `
+apiVersion: example.com/v1beta1
+kind: Example
+spec:
+  list: # {"$kpt-set":"${list}"}
+  - 0
+ `,
+			expectedOpenAPI: `
+kind: Kptfile
+openAPI:
+  definitions:
+    io.k8s.cli.setters.list:
+      type: array
+      x-k8s-cli:
+        setter:
+          name: list
+          listValues:
+          - "10"
+          - "11"
+          isSet: true
+ `,
+			expectedResources: `
+apiVersion: example.com/v1beta1
+kind: Example
+spec:
+  list: # {"$kpt-set":"${list}"}
+  - "10"
+  - "11"
+ `,
+		},
+	}
+	for i := range tests {
+		test := tests[i]
+		t.Run(test.name, func(t *testing.T) {
+			// reset the openAPI afterward
+			openapi.ResetOpenAPI()
+			defer openapi.ResetOpenAPI()
```

Comment:
Is this still needed? Are we still mutating the openapi schema?

Reply:
Not needed. Removed.